autoload/codefmt/shfmt.vim

```diff
@@ -0,0 +1,91 @@
+" Copyright 2017 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+
+let s:plugin = maktaba#plugin#Get('codefmt')
+
+
+""
+" @private
+" Formatter: shfmt
+function! codefmt#shfmt#GetFormatter() abort
+  let l:formatter = {
+      \ 'name': 'shfmt',
+      \ 'setup_instructions': 'Install shfmt and configure the ' .  
```

Comment:
This should link to where you can get shfmt or find out more about it. There's nothing else to disambiguate if another tool called shfmt comes along, whereas the code is crafted to work with that particular tool.